Use std::vectory instead of malloc in solver allocation

payetvin opened the pull request:
close #1304

payetvin committed:
[DEV] Replace memalloc by new in sim_alloc

flomnes reviewed (changes requested):
Don't use `free` in this case, use `delete` or `delete[]`

payetvin committed:
[DEV] Replace free by delete in sim_alloc

payetvin committed:
[DEV] Fix valgrind errors

payetvin committed:
[DEV] Bool as vectors

payetvin committed:
[DEV] CoefficientEcretementPMaxHydraulique and BruitSurCoutHydraulique

payetvin committed:
[DEV] int index, numero and nbgrp

payetvin committed:
[DEV] double coutdedefaillance

payetvin committed:
[DEV] int numerodecontraite

flomnes commented on src/solver/simulation/sim_alloc_probleme_hebdo.cpp:
```diff
@@ -46,70 +46,64 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, uint NombreDePasDeTemp
-    problem.DefaillanceNegativeUtiliserPMinThermique = (bool*)MemAlloc(nbPays * sizeof(char));
-    problem.DefaillanceNegativeUtiliserHydro = (bool*)MemAlloc(nbPays * sizeof(char));
-    problem.DefaillanceNegativeUtiliserConsoAbattue = (bool*)MemAlloc(nbPays * sizeof(char));
+    problem.DefaillanceNegativeUtiliserPMinThermique = std::vector<bool>(nbPays);
```
```suggestion
    problem.DefaillanceNegativeUtiliserPMinThermique.assign(nbPays, false);
```

payetvin committed:
[FIX] Tests error

payetvin committed:
[DEV] more int

payetvin committed:
[DEV] NumeroDeVariableDeTrancheDeStock + assign for vector<vector<>>

payetvin committed:
[DEV] COUTS_DE_TRANSPORT

payetvin committed:
[DEV] assign, previous

payetvin committed:
[DEV] ValeursDeNTC start

payetvin committed:
[DEV] ValeursDeNTC in progress

payetvin committed:
[DEV] ValeursDeNTC compiling

payetvin committed:
[DEV] ConsommationsAbattues

payetvin committed:
[DEV] AllMustRunGeneration

payetvin committed:
[DEV] SoldeMoyenHoraire

guilpier-code commented on src/solver/simulation/sim_alloc_probleme_hebdo.cpp:
```diff
@@ -46,307 +46,270 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, uint NombreDePasDeTemp
+
+    problem.BruitSurCoutHydraulique.assign(nbPays, std::vector<double>(8784));
+
+    problem.NomsDesPays = std::vector<const char*>(nbPays);
```
About : `problem.NomsDesPays = std::vector<const char*>(nbPays);`
Why not : 
```C++
problem.NomsDesPays.resize(nbPays);
```
? 
Same remark for other assignment like this

flomnes replied:
I agree, please use `assign` for consistency and to make sure memory is initialized.

payetvin committed:
[DEV] CORRESPONDANCES_DES_VARIABLES

payetvin committed:
[DEV] std::vector<CORRESPONDANCES_DES_CONTRAINTES*>

payetvin committed:
[DEV] vector in struct CORRESPONDANCES_DES_CONTRAINTES

payetvin committed:
[DEV] CORRESPONDANCES_DES_CONTRAINTES only vectors

payetvin committed:
[DEV] RESERVE_JMOINS1 done

payetvin committed:
[DEV] Use resize

payetvin committed:
[DEV] std::vector<VARIABLES_DUALES_INTERCONNEXIONS*>

payetvin committed:
[DEV] std::vector<double> VariableDualeParInterconnexion

payetvin committed:
[DEV] std::vector<VARIABLES_DUALES_INTERCONNEXIONS> VariablesDualesDesContraintesDeNTC

payetvin committed:
[DEV] std::vector<CONTRAINTES_COUPLANTES*> MatriceDesContraintesCouplantes

payetvin committed:
[DEV] vector in struct CONTRAINTES_COUPLANTES

payetvin committed:
[DEV] mutable std::vector<CONTRAINTES_COUPLANTES> MatriceDesContraintesCouplantes

payetvin committed:
[DEV] std::vector<PALIERS_THERMIQUES*> PaliersThermiquesDuPays

payetvin committed:
[DEV] vector in struct PALIERS_THERMIQUES

payetvin committed:
[DEV]  std::vector<PALIERS_THERMIQUES> PaliersThermiquesDuPays

payetvin committed:
[DEV] std::vector<ENERGIES_ET_PUISSANCES_HYDRAULIQUES*> CaracteristiquesHydrauliques

payetvin committed:
[DEV] vector in struct ENERGIES_ET_PUISSANCES_HYDRAULIQUES

payetvin committed:
[DEV] std::vector<ENERGIES_ET_PUISSANCES_HYDRAULIQUES> CaracteristiquesHydrauliques

payetvin committed:
[DEV] std::vector<COUTS_MARGINAUX_ZONES_DE_RESERVE*> CoutsMarginauxDesContraintesDeReserveParZone

payetvin committed:
[DEV] vector in struct COUTS_MARGINAUX_ZONES_DE_RESERVE

payetvin committed:
[DEV] std::vector<COUTS_MARGINAUX_ZONES_DE_RESERVE> CoutsMarginauxDesContraintesDeReserveParZone

payetvin committed:
[DEV] std::vector<CORRESPONDANCES_DES_CONTRAINTES_JOURNALIERES> CorrespondanceCntNativesCntOptimJournalieres;
    CORRESPONDANCES_DES_CONTRAINTES_HEBDOMADAIRES CorrespondanceCntNativesCntOptimHebdomadaires;

payetvin committed:
[DEV] coutOptimalSolution1, coutOptimalSolution2, tempsResolution1, tempsResolution2

payetvin committed:
[DEV] Remove CoutsMarginauxDesContraintesDeReserveParZone

payetvin committed:
[DEV] remove struct COUTS_MARGINAUX_ZONES_DE_RESERVE

payetvin committed:
[DEV] remove struct COUTS_MARGINAUX_ZONES_DE_RESERVE from solver.hxx

payetvin committed:
[DEV] Try fixing windows error

payetvin committed:
[MERGE] develop

flomnes commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -561,58 +548,56 @@ struct PROBLEME_HEBDO
-    CORRESPONDANCES_DES_CONTRAINTES** CorrespondanceCntNativesCntOptim = nullptr;
-    CORRESPONDANCES_DES_CONTRAINTES_JOURNALIERES** CorrespondanceCntNativesCntOptimJournalieres = nullptr;
-    CORRESPONDANCES_DES_CONTRAINTES_HEBDOMADAIRES** CorrespondanceCntNativesCntOptimHebdomadaires = nullptr;
+    std::vector<CORRESPONDANCES_DES_VARIABLES*> CorrespondanceVarNativesVarOptim;
```
Please leave a TODO note here, and also an explanation on what's not working

flomnes commented on src/solver/simulation/sim_alloc_probleme_hebdo.cpp:
```diff
@@ -46,295 +46,253 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, unsigned NombreDePasDe
+
+    problem.CoefficientEcretementPMaxHydraulique.assign(nbPays, 0.);
+
+    problem.BruitSurCoutHydraulique.assign(nbPays, std::vector<double>(8784));
```
Use named constants, not hard-coded numbers.

flomnes commented on src/solver/simulation/common-hydro-levels.cpp:
```diff
@@ -62,15 +63,13 @@ void computingHydroLevels(const Data::AreaList& areas,
-
-        computeLvlObj.setParameters(
-          nivInit, inflows, ovf, turb, pumpingRatio, pump, reservoirCapacity);
+        auto computeLvlObj = computeTimeStepLevel(
```
```suggestion
        computeTimeStepLevel computeLvlObj(
```

flomnes reviewed (changes requested):
LGTM. Only minor changes requested.

flomnes commented:
Please also leave a TODO note for structures used to store results.

payetvin committed:
[DEV] Comments

payetvin committed:
[FIX] revert HOURS_PER_YEAR for BruitSurCoutHydraulique

flomnes committed:
Style

flomnes commented on src/solver/simulation/solver.hxx:
```diff
@@ -146,7 +146,7 @@ private:
 
             if (hydroHotStart && firstSetParallelWithAPerformedYearWasRun)
-                randomReservoirLevel = state[numSpace].problemeHebdo->previousYearFinalLevels;
+                randomReservoirLevel = &state[numSpace].problemeHebdo->previousYearFinalLevels[0];
```
```suggestion
                randomReservoirLevel = state[numSpace].problemeHebdo->previousYearFinalLevels.data();
```

flomnes commented on src/solver/simulation/sim_alloc_probleme_hebdo.cpp:
```diff
@@ -580,259 +535,82 @@ void SIM_DesallocationProblemeHebdo(PROBLEME_HEBDO& problem)
-                      ->PuissanceDisponibleEtCout[j]
-                      ->CoutHoraireDuPalierThermiqueDown;
-            delete problem.PaliersThermiquesDuPays[k]->PuissanceDisponibleEtCout[j];
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->CoutHoraireDeProductionDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->CoutHoraireDeProductionDuPalierThermiqueRef);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceDisponibleDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceDisponibleDuPalierThermiqueRef);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceDisponibleDuPalierThermiqueRef_SV);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceMinDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceMinDuPalierThermique_SV);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->NombreMaxDeGroupesEnMarcheDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->NombreMinDeGroupesEnMarcheDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->CoutHoraireDuPalierThermiqueUp);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
```
Be careful, you're mixing `new` (C++) and `free` (C)

payetvin replied:
Will be deleted when replaced with vectors

payetvin committed:
[DEV] std::vector<PDISP_ET_COUTS_HORAIRES_PAR_PALIER*> PuissanceDisponibleEtCout

payetvin committed:
[DEV] struct PDISP_ET_COUTS_HORAIRES_PAR_PALIER

payetvin committed:
[DEV] std::vector<PDISP_ET_COUTS_HORAIRES_PAR_PALIER> PuissanceDisponibleEtCout

payetvin committed:
[DEV] std::vector<RESULTATS_CONTRAINTES_COUPLANTES> ResultatsContraintesCouplantes;

payetvin committed:
[DEV] struct RESULTATS_HORAIRES

payetvin committed:
[DEV] Fixing tests errors

payetvin committed:
[DEV] std::vector<PRODUCTION_THERMIQUE_OPTIMALE*> ProductionThermique;

payetvin committed:
[DEV] struct PRODUCTION_THERMIQUE_OPTIMALE

payetvin committed:
[DEV] std::vector<PRODUCTION_THERMIQUE_OPTIMALE> ProductionThermique;

payetvin committed:
[MERGE] Fix conflicts

payetvin committed:
[MERGE] Fix var reference

payetvin committed:
[FIX] linkCount into clusterCount

payetvin committed:
[CI] Commented 860 to validate other tests

payetvin committed:
[CI] Revert 860
